tests/FSharpLint.Core.Tests/Rules/Binding/FavourAsKeyword.fs

```diff
@@ -0,0 +1,42 @@
+module FSharpLint.Core.Tests.Rules.Binding.FavourAsKeyword
+
+open NUnit.Framework
+open FSharpLint.Rules
+
+[<TestFixture>]
+type TestBindingFavourAsKeyword() =
+    inherit TestAstNodeRuleBase.TestAstNodeRuleBase(FavourAsKeyword.rule)
+
+    [<Test>]
+    member this.FavourAsKeywordShouldQuickFixAndProduceError() =
+        let source = """
+module Program
+
+match "" with
+| bar when bar = "baz" -> ()
+"""
+
+        this.Parse(source)
+
+        let expected = """
+module Program
+
+match "" with
+| "baz" as bar -> ()
+"""
+
+        Assert.AreEqual(expected, this.ApplyQuickFix source)
+        this.AssertErrorWithMessageExists("Prefer using the as pattern to match a constant and bind it to a variable.")
```

Comment:
@ribeirotomas1904 let's rather have two tests: one for the AssertErrorWithMessageExists and another one for the QuickFix

Reply:
done in 573f0e06227360bfc6e107ab7fd863b6691a5ea7